magicblock-committor-service/src/intent_executor/two_stage_executor.rs

```
@@ -0,0 +1,221 @@
+        let finalize_signature = finalize_result.map_err(|err| {
+            IntentExecutorError::from_strategy_execution_error(
+                err,
+                |internal_err| {
+                    let finalize_signature = internal_err.signature();
+                    IntentExecutorError::FailedToFinalizeError {
+                        err: internal_err,
+                        commit_signature: Some(commit_signature),
+                        finalize_signature,
+                    }
+                },
+            )
+        })?;
+
+        Ok(ExecutionOutput::TwoStage {
+            commit_signature,
+            finalize_signature,
+        })
+    }
+
+    pub async fn patch_commit_strategy(
+        &self,
+        err: &TransactionStrategyExecutionError,
+        commit_strategy: &mut TransactionStrategy,
+        committed_pubkeys: &[Pubkey],
+    ) -> IntentExecutorResult<ControlFlow<(), TransactionStrategy>> {
+        match err {
+            TransactionStrategyExecutionError::CommitIDError => {
+                let to_cleanup = self
+                    .handle_commit_id_error(committed_pubkeys, commit_strategy)
+                    .await?;
+                Ok(ControlFlow::Continue(to_cleanup))
+            }
+            TransactionStrategyExecutionError::ActionsError => {
+                // Unexpected in Two Stage commit
+                // That would mean that Two Stage executes Standalone commit
+                warn!("Unexpected error in Two stage commit flow: {}", err);
+                Ok(ControlFlow::Break(()))
+            }
+            TransactionStrategyExecutionError::CpiLimitError => {
+                // Can't be handled
+                warn!("Commit tasks exceeded CpiLimitError: {}", err);
+                Ok(ControlFlow::Break(()))
+            }
+            TransactionStrategyExecutionError::InternalError(_) => {
+                // Can't be handled
+                Ok(ControlFlow::Break(()))
```

Swallowing the inner error + not logging anything will make this impossible to triage if this ever happens.